common/pkg/server/server.go

```diff
@@ -83,7 +83,7 @@ func (s *Server) Start(ctx context.Context) error {
 		h := s.hooks[idx]
 
 		if t, ok := h.(lifecycle.BeforeClusterCreate); ok {
-			if err := webhookServer.AddExtensionHandler(server.ExtensionHandler{
+			if err := webhookServer.AddExtensionHandler(runtimeserver.ExtensionHandler{
 				Hook:        runtimehooksv1.BeforeClusterCreate,
 				Name:        strings.ToLower(h.Name()),
```

Comment:
We will need to add suffixes to every handler type
```suggestion
				Name:        strings.ToLower(h.Name()+"-bcc"),
```